src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1218,3 +1221,20 @@
     (when (fs/exists? cert)
       (fs/delete cert)
       (log/debug "Deleted certificate for" subject))))
+
+(schema/defn ^:always-validate get-custom-oid-mappings :- (schema/maybe OIDMappings)
+  "Given a path to a custom OID mappings file, return a map of all oids to
+  shortnames"
+  [custom-oid-mapping-file :- (schema/maybe schema/Str)]
+  (when (fs/file? custom-oid-mapping-file)
+    (try
+      (let [oid-mappings (:oid_mapping (yaml/parse-string (slurp custom-oid-mapping-file)))]
+        (into {} (for [[oid names] oid-mappings] [(name oid) (keyword (:shortname names))])))
+      (catch Exception e
```

Comment:
`catch Exception` here looks a bit suspicious.  Does `parse-string` throw something more specific that could be caught instead?  And, actually, could we just let that exception bubble-out instead of this `try`/`catch`/`throw`?